syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -145,6 +172,18 @@ def backward(self, *args, **kwargs):
         we just ignore the call."""
         pass
 
+    def modulo(self, x, field):
+        mask_pos = x > ((field - 1) // 2)
+        mask_neg = x < -(field // 2)
+        if mask_pos.any():
+            mask_pos = mask_pos.long()
+            return self.modulo(x - (mask_pos * field), field)
+        elif mask_neg.any():
+            mask_neg = mask_neg.long()
+            return self.modulo(x + (mask_neg * field), field)
+        else:
+            return x.type(self.field_to_torch_dtype)
```

Comment:
Yes, for automatic modulo based on native tensor dtype.

Reply:
I just find it a bit weird that this function is applied to x, takes a field and uses self for the dtype (which should be possible to infer from the field, am I wrong?).

Reply:
Yes, that does seem like a roundabout way of doing it and it can cause if not issues then confusion in the future. I will modify it to derive torch_dtype from the field itself here.

Reply:
And then, if self is not used, you can make this a static method :)

Reply:
Yes that will be for the best. As I can revert generate_shares to be static too then.